### PR TITLE
Fix table tests

### PR DESCRIPTION
**Pull request by themadcreator**
- Call super methods of AbstractComponent instead of validateProps directly
- Use new enzyme state method
- Use full mount instead of shallow mount
- Remove update tests due to awkward error boundary prop setting with new enzyme API (also already tested on mount)

### CODE REVIEW

File: packages/table/package.json [hunk 1]
```diff
@@ -24,7 +24,7 @@
     "test": "npm-run-all -s compile:cjs test:pre -p test:karma test:iso",
     "test:pre": "tsc -p ./test",
     "test:karma": "karma start",
-    "test:karma:debug": "karma start --browsers=Chrome --single-run=false --reporters=mocha --debug",
+    "test:karma:debug": "karma start --single-run=false --reporters=mocha --debug",
```
Comment (gscshoyru) [inline packages/table/package.json]: This probably shouldn't stay in, yes?

File: packages/table/test/tableTests.tsx [hunk 2]
```diff
@@ -389,15 +389,32 @@ describe("<Table>", () => {
 
         it("Selects and deselects column/row headers when selecting and deselecting the full table", () => {
             const table = mountTable();
-            const columnHeader = table.find(COLUMN_HEADER_SELECTOR).at(0);
-            const rowHeader = table.find(`.${Classes.TABLE_ROW_HEADERS} .${Classes.TABLE_HEADER}`).at(0);
 
+            // selet the full table
```
Comment (giladgray) [inline packages/table/test/tableTests.tsx]: # TYPO

File: packages/table/test/tableTests.tsx [hunk 3]
```diff
@@ -389,15 +389,32 @@ describe("<Table>", () => {
...
             selectFullTable(table);
+            let columnHeader = table
+                .find(COLUMN_HEADER_SELECTOR)
+                .hostNodes()
+                .at(0);
+            let rowHeader = table
+                .find(`.${Classes.TABLE_ROW_HEADERS}`)
+                .find(`.${Classes.TABLE_HEADER}`)
+                .hostNodes()
+                .at(0);
```
Comment (giladgray) [inline packages/table/test/tableTests.tsx]: `.first()`?

File: packages/table/test/tableTests.tsx [hunk 4]
```diff
@@ -389,15 +389,32 @@ describe("<Table>", () => {
...
 
+            // selet the full table
             selectFullTable(table);
+            let columnHeader = table
+                .find(COLUMN_HEADER_SELECTOR)
+                .hostNodes()
+                .at(0);
+            let rowHeader = table
+                .find(`.${Classes.TABLE_ROW_HEADERS}`)
+                .find(`.${Classes.TABLE_HEADER}`)
```
Comment (giladgray) [inline packages/table/test/tableTests.tsx]: can combine these into a compound selector: `.find(".one .two")`
(few other places too)

**Pull request merged**